presto-native-execution/presto_cpp/main/LinuxMemoryChecker.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/regex.hpp>
+#include <folly/Conv.h>
+#include <folly/CppAttributes.h>
+#include <folly/File.h>
+#include <folly/FileUtil.h>
+#include <folly/Format.h>
+#include <folly/Range.h>
+#include <folly/String.h>
+#include <folly/gen/Base.h>
+#include <folly/gen/File.h>
+#include <folly/gen/String.h>
+#include "presto_cpp/main/PeriodicMemoryChecker.h"
+#include "presto_cpp/main/common/Configs.h"
+
+#include <sys/stat.h>
+
+namespace facebook::presto {
+
+class LinuxMemoryChecker : public PeriodicMemoryChecker {
+ public:
+  explicit LinuxMemoryChecker(
+      const PeriodicMemoryChecker::Config& config,
+      int64_t mallocBytes = 0,
+      std::function<void()>&& periodicCb = nullptr,
+      std::function<bool(const std::string&)>&& heapDumpCb = nullptr)
+      : PeriodicMemoryChecker(config),
+        mallocBytes_(mallocBytes),
+        periodicCb_(std::move(periodicCb)),
+        heapDumpCb_(std::move(heapDumpCb)) {}
+
+  ~LinuxMemoryChecker() override {}
+
+  void setMallocBytes(int64_t mallocBytes) {
+    mallocBytes_ = mallocBytes;
+  }
+
+  static std::unique_ptr<PeriodicMemoryChecker> createLinuxMemoryChecker() {
```

Comment:
This can be in an anonymous namespace.

Reply:
Moved createLinuxMemoryChecker() to anonymous namespace